rclcpp/src/rclcpp/node_interfaces/node_parameters.cpp

```diff
@@ -222,6 +222,31 @@ __check_parameter_value_in_range(
     return result;
   }
 
+  if (!descriptor.integer_range.empty() && value.get_type() == rclcpp::PARAMETER_INTEGER_ARRAY) {
+    std::vector<int64_t> v = value.get<std::vector<int64_t>>();
+    auto integer_range = descriptor.integer_range.at(0);
+    for (const int64_t & val : v) {
+      if (val == integer_range.from_value || val == integer_range.to_value) {
```

Comment:
please refactor to use a common code with the single integer part.

Reply:
agree, that inner function can be called in this for loop.

Reply:
Done